sway-lib-std/src/storage.sw

```diff
@@ -681,6 +681,262 @@ impl<V> StorageVec<V> {
     pub fn clear(self) {
         let _ = clear::<u64>(__get_storage_key());
     }
+
+    /// Swaps two elements.
+    ///
+    /// ### Arguments
+    ///
+    /// * element1_index - The index of the first element.
+    /// * element2_index - The index of the second element.
+    ///
+    /// ### Reverts
+    ///
+    /// * If `element1_index` or `element2_index` is greater than the length of the vector.
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads: `3`
+    /// * Writes: `2`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std::storage::StorageVec;
+    ///
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {}
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///
+    ///     storage.vec.swap(0, 2);
```

Comment:
I would assert the length and capacity before and after the call for pre/post conditional completeness.